healpy/sphtfunc.py

```diff
@@ -224,18 +230,29 @@ def map2alm(
     nside = pixelfunc.get_nside(maps)
     check_max_nside(nside)
 
+    pixel_weights_filename = None
     if use_pixel_weights:
         if use_weights:
             raise RuntimeError("Either use pixel or ring weights")
-        with data.conf.set_temp("dataurl", DATAURL), data.conf.set_temp(
-            "dataurl_mirror", DATAURL_MIRROR
-        ), data.conf.set_temp("remote_timeout", 30):
-            pixel_weights_filename = data.get_pkg_data_filename(
-                "full_weights/healpix_full_weights_nside_%04d.fits" % nside,
-                package="healpy",
-            )
-    else:
-        pixel_weights_filename = None
+        filename = "full_weights/healpix_full_weights_nside_%04d.fits" % nside
+        if datapath is not None:
+            pixel_weights_filename = os.path.join(datapath, filename)
+            if os.path.exists(pixel_weights_filename):
+                warnings.warn(
```

Comment:
This could potentially generate many lines of output if `map2alm()` is called hundreds of times (e.g. by TOAST doing coordinate rotation once per detector).  However, we can filter out healpy warnings if needed, so that is fine.